### PR TITLE
[homekit] Fix Eclipse warnings and reduce bogus logger ERRORs to DEBUGs.

### LINKED ISSUE #1258: [homekit] Duplicate subscription
Hi,

After installing the binding and testing it I got the following ERRORs:

[ERROR] [kit.internal.HomekitAccessoryUpdater] - Received duplicate subscription on XXXXXXX

for many of my Items how can I fix this?

best regards René
##

### PR DESCRIPTION
**Pull request by 9037568**
Probably fixes #1258, since most of the descriptions given say the binding is functioning correctly in spite of seeing bogus duplicate subscription messages.

See also this [community thread](https://community.openhab.org/t/homekit-binding/37690/14).

Signed-off-by: Chris Carman <namraccr@gmail.com>

### CODE REVIEW

File: addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryRegistry.java [hunk 1]
```diff
@@ -85,12 +85,12 @@ public synchronized void addCharacteristic(HomekitTaggedItem item) {
...
             if (pendingGroupedAccessories.containsKey(group)) {
                 addCharacteristicToGroup(group, item);
-                logger.debug("Added {} to {}", item.getItem().getName(), group);
+                logger.debug("Added item {} to group {}", item.getItem().getName(), group);
                 return;
             }
         }
         pendingCharacteristics.add(item);
-        logger.debug("Stored {} until group is ready", item.getItem().getName());
+        logger.debug("Stored item{} until group is ready", item.getItem().getName());
```
Comment (cweitkamp) [inline addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryRegistry.java]: Missing space.

```suggestion
        logger.debug("Stored item {} until group is ready", item.getItem().getName());
```

File: addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryUpdater.java [hunk 2]
```diff
@@ -47,15 +47,16 @@ public void subscribe(GenericItem item, String key, HomekitCharacteristicChangeC
...
         }
         subscriptionsByName.compute(itemKey, (k, v) -> {
             if (v != null) {
-                logger.error("Received duplicate subscription on {}", item.getName());
+                logger.debug("Compute: received duplicate subscription on item {} for key {}. Will unsubscribe.", item.getName(), key);
                 unsubscribe(item, key);
             }
             Subscription subscription = (changedItem, oldState, newState) -> callback.changed();
             item.addStateChangeListener(subscription);
+            logger.debug("Successfully added subscription for item '{}' using key '{}'", item.getName(), key);
```
Comment (cweitkamp) [inline addons/io/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryUpdater.java]: Without quotes like the rest of the logging messages.

```suggestion
            logger.debug("Successfully added subscription for item {} using key {}", item.getName(), key);
```

**Comment by cweitkamp** [comment]
@9037568 Did you see my comments? Do you want to apply the suggestions or not?

**Comment by davidgraeff** [comment]
Because this will lead to a merge conflict when migrated, I'll merge it now.
I'm sure someone will be annoyed by the missing whitespace and fix that at some point.

**Pull request merged**